### PR TITLE
Switch to UV from PDM

### PR DESCRIPTION
**Pull request by ewilliams-cloudera**

### CODE REVIEW

File: .github/workflows/llm_checks.yml [hunk 1]
```diff
@@ -14,37 +14,35 @@ jobs:
...
+        run: astral-sh/setup-uv@v3
+        with:
+          enable-cache: true
+          cache-local-path: "./llm-service/uv.lock"
+
       - name: Set up Python
         uses: pdm-project/setup-pdm@v3
         with:
           python-version: '3.10'
           cache: false
```
Comment (conradocloudera) [inline .github/workflows/llm_checks.yml:27]: remove?

File: .github/workflows/llm_checks.yml [hunk 2]
```diff
@@ -14,37 +14,31 @@ jobs:
...
+          enable-cache: true
+          cache-local-path: "./llm-service/uv.lock"
 
       - name: Install Python dependencies
-        if: steps.python_cache.outputs.cache-hit != 'true'
         run: |
-          pdm install
+          uv python install 3.10
+          uv venv
+          uv pip install -r pyproject.toml
```
Comment (conradocloudera) [inline .github/workflows/llm_checks.yml:27]: can we install from the locked versions instead?

File: local-dev.sh [hunk 3]
```diff
@@ -65,11 +65,11 @@ docker run --name qdrant_dev --rm -d -p 6333:6333 -p 6334:6334 -v $(pwd)/databas
 cd llm-service
 python3.10 -m venv venv
 source venv/bin/activate
-python -m pip install pdm
-pdm install
-pdm run pytest -sxvvra
+python -m pip install uv
+uv pip install -r pyproject.toml
```
Comment (conradocloudera) [inline local-dev.sh:69]: Same

File: scripts/refresh_project.sh [hunk 4]
```diff
@@ -46,8 +46,9 @@ cd ui/express
 npm install
 
 cd ../../llm-service
-pip install pdm
-pdm install -v
+pip install uv
+uv venv
+uv pip install -r pyproject.toml
```
Comment (conradocloudera) [inline scripts/refresh_project.sh:51]: Same

File: llm-service/Dockerfile [hunk 5]
```diff
@@ -1,9 +1,9 @@
 FROM docker-private.infra.cloudera.com/cloudera_base/hardened/cloudera-python:3.10
-RUN pip install pdm
+RUN pip install uv
 COPY ./pyproject.toml /app/
-COPY ./pdm.lock /app/
+COPY ./uv.lock /app/
 WORKDIR /app
-RUN pdm install
+RUN uv install
```
Comment (conradocloudera) [inline llm-service/Dockerfile:6]: The other places used `uv pip install -r pyproject.toml`. Why is this different? 🤔

File: .github/workflows/llm_checks.yml (hunk 2 above)
Comment (ewilliams-cloudera) [reply .github/workflows/llm_checks.yml:27]: Maybe?  Let me see if that's possible

**Review by jkwatson — APPROVED** [review]
🚢

**Pull request merged**